test/source/tests/settings.ts

```diff
@@ -706,6 +706,45 @@ export const defineSettingsTests = (testVariant: TestVariant, testWithBrowser: T
         internalTestState.expectIntentionalErrReport = true;
       })
     );
+    test(
+      'settings - Errors encountered during the addition/update of a private key should be handled properly',
+      testWithBrowser(async (t, browser) => {
+        t.context.mockApi!.configProvider = new ConfigurationProvider({
+          attester: {
+            pubkeyLookup: {},
+          },
+        });
+        const acct = 'flowcrypt.compatibility@gmail.com';
+        const settingsPage = await BrowserRecipe.openSettingsLoginApprove(t, browser, acct);
+        const keyCanBeFixedTestKey = testConstants.keyCanBeFixedTestKey;
+        await SetupPageRecipe.manualEnter(settingsPage, '', {
+          fixKey: true,
+          key: {
+            title: '',
+            armored: keyCanBeFixedTestKey,
+            passphrase: 'hard to guess passphrase',
+            longid: '9866DB9063926D73',
+          },
+        });
+        await SettingsPageRecipe.toggleScreen(settingsPage, 'additional');
+        await settingsPage.waitAndClick('@action-open-pubkey-page');
+        const myKeyPage = await settingsPage.getFrame(['my_key.htm']);
+        await myKeyPage.waitAndClick('@action-update-prv');
+        await myKeyPage.waitAndClick('@source-paste');
+        const invalidKeyFormat = 'non-valid-private-key-format';
+        const invalidPrivateKeyBlock = '-----BEGIN PGP PRIVATE KEY BLOCK-----\r\n\r\ninvalid-key\r\n-----END PGP PRIVATE KEY BLOCK-----';
+        await myKeyPage.waitAndType('@input-prv-key', invalidKeyFormat);
+        await myKeyPage.waitAndClick('@action-update-key');
+        await myKeyPage.waitAndRespondToModal('warning', 'confirm', 'This does not appear to be a validly formatted key.');
+        await myKeyPage.waitAndType('@input-prv-key', invalidPrivateKeyBlock);
+        await myKeyPage.waitAndClick('@action-update-key');
+        await myKeyPage.waitAndRespondToModal('error', 'confirm', 'An error happened when processing the key');
+        await myKeyPage.waitAndType('@input-prv-key', keyCanBeFixedTestKey);
+        await myKeyPage.waitAndClick('@action-update-key');
+        await browser.newExtensionPage(t, 'chrome/settings/setup.htm');
+        await settingsPage.close();
```

Comment:
why do we need to open another page here, probably these 2 lines can be deleted?

Reply:
ah yes - it can be deleted. thanks!